pkg/trustedresources/verify.go

```diff
@@ -169,32 +175,62 @@ func getMatchedPolicies(resourceName string, source string, policies []*v1alpha1
 //  1. If multiple policies match, the resource must satisfy all the "enforce" policies to pass verification. The matching "enforce" policies are evaluated using AND logic.
 //     Alternatively, if the resource only matches policies in "warn" mode, it will still pass verification and only log a warning if these policies are not satisfied.
 //  2. To pass one policy, the resource can pass any public keys in the policy. We use OR logic on public keys of one policy.
-func verifyResource(ctx context.Context, resource metav1.Object, k8s kubernetes.Interface, signature []byte, matchedPolicies []*v1alpha1.VerificationPolicy) error {
+//
+// TODO(#6683): return all failed policies in error.
+func verifyResource(ctx context.Context, resource metav1.Object, k8s kubernetes.Interface, signature []byte, matchedPolicies []*v1alpha1.VerificationPolicy) VerificationResult {
+	logger := logging.FromContext(ctx)
+	var warnPolicies []*v1alpha1.VerificationPolicy
+	var enforcePolicies []*v1alpha1.VerificationPolicy
 	for _, p := range matchedPolicies {
-		passVerification := false
+		if p.Spec.Mode == v1alpha1.ModeWarn {
+			warnPolicies = append(warnPolicies, p)
+		} else {
+			enforcePolicies = append(enforcePolicies, p)
+		}
+	}
+
+	// first evaluate all enforce policies. Return VerificationError type of VerificationResult if any policy fails.
+	for _, p := range enforcePolicies {
 		verifiers, err := verifier.FromPolicy(ctx, k8s, p)
 		if err != nil {
-			return fmt.Errorf("failed to get verifiers from policy: %w", err)
+			return VerificationResult{VerificationResultType: VerificationError, Err: fmt.Errorf("failed to get verifiers from policy: %w", err)}
 		}
-		for _, verifier := range verifiers {
-			// if one of the verifier passes verification, then this policy passes verification
-			if err := verifyInterface(resource, verifier, signature); err == nil {
-				passVerification = true
-				break
-			}
+		passVerification := doesAnyVerifierPass(resource, signature, verifiers)
+		if !passVerification {
+			return VerificationResult{VerificationResultType: VerificationError, Err: fmt.Errorf("%w: resource %s in namespace %s fails verification", ErrResourceVerificationFailed, resource.GetName(), resource.GetNamespace())}
 		}
-		// if this policy fails the verification and the mode is not "warn", should return error directly. No need to check other policies
+	}
+
+	// return VerificationWarn type of VerificationResult if any warn policies fails.
```

Comment:
Just to be consistent.
```suggestion
	// then evaluate all warn policies. Return VerificationWarn type of VerificationResult if any warn policies fails.
```

Reply:
Thanks!